newrelic/structures_newrelic_one_dashboard_json_test.go

```diff
@@ -0,0 +1,44 @@
+package newrelic
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/newrelic/newrelic-client-go/v2/pkg/dashboards"
+)
+
+func TestSetDataAccounts(t *testing.T) {
+	const fixtureInName = "dashboard_data_account.json"
+	const fixtureOutName = "dashboard_data_account_expected.json"
+
+	inputFile, err := os.ReadFile(filepath.Join("testdata", fixtureInName))
+	if err != nil {
+		t.Fatalf("read fixture: %v", err)
+	}
+	dash := dashboards.DashboardInput{}
+	if err := json.Unmarshal(inputFile, &dash); err != nil {
+		t.Fatalf("unmarshal fixture: %v", err)
+	}
+
+	dataAccounts := []any{456, 789}
+	if err := setDataAccounts(&dash, dataAccounts); err != nil {
+		t.Fatalf("setDataAccounts: %v", err)
+	}
```

Comment:
```suggestion
	dash := dashboards.DashboardInput{}
	if unmarshalErr := json.Unmarshal(inputFile, &dash); unmarshalErr != nil {
		t.Fatalf("unmarshal fixture: %v", err)
	}

	dataAccounts := []any{456, 789}
	if setDataAccountsErr := setDataAccounts(&dash, dataAccounts); setDataAccountsErr != nil {
		t.Fatalf("setDataAccounts: %v", err)
	}
```

(since the lint failures show the variable `err` is getting masked by multiple references to the same)

Reply:
Updated it. Instead of renaming it, I re-assigned to the existing variable because a custom name for an `err` leads to easy mistakes like showing the wrong `err` in the error message like here: `t.Fatalf("unmarshal fixture: %v", err)`

Reply:
Agreed, thanks. Lint failures are now gone 👍